### ISSUE TITLE
How to use with Apollo?

### ISSUE THREAD

**Issue by joncursi** [exponentjs/react-native-loading-container#15]
I've been using this library with great success for simple fetch requests to external resources, but now I'm bringing Apollo / GraphQL into my codebase and I'm wondering how to properly use it with this package.

Apollo fetches data using its own HOC. You specify the query, and the data arrives *at some point*

```jsx
const withQuery = graphql(gql`
  {
    nativeCurrencies {
      name
      symbol
    }
  }
`);

const enhance = compose(
  withQuery,
);

export default enhance(SomeComponent);
```

So my logic for handing the fetch is abstracted away into Apollo's HOC, so I don't have direct control over it anymore. All I have to work with is whether or not `data.loading` is true or false, which is great for DX. But I start to lose control over things like this.

Is there a way to get these two systems to work together? Perhaps, passing in my loading state directly rather than specifying the async functions?

**Comment by joncursi** [comment]
cc @stubailo

**Comment by brentvatne** [comment]
probably there is no need to use this with apollo because apollo already provides the same sort of behavior, minus the activity indicator :)

**Comment by stubailo** [comment]
Yeah, I think you can display a loading indicator based on that field, would it be helpful to have a special component for that? You can see an example on the Apollo Client homepage, which uses an ActivityIndicator for that.
